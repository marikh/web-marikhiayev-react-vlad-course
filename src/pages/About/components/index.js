import AboutHeroArea from './AboutHeroArea';

export {
    AboutHeroArea
}
