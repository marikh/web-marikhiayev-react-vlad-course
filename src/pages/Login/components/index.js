import LoginHeroArea from './LoginHeroArea';

export {
    LoginHeroArea
}
