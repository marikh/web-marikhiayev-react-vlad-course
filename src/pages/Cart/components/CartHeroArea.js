import React from 'react';
import { HeroArea, Heading } from '../../../components';

export default () => (
    <HeroArea>
        <Heading size={3}>
            Cart
        </Heading>
    </HeroArea>
) 