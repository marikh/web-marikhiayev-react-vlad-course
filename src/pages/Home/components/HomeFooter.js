import React from 'react';
import { SocialLinks } from '../../../components/';
import { withRouter } from 'react-router-dom';

export default withRouter(({ links, history, location, match }) => {
    return (
        <div>
            <SocialLinks links={links} />
            <h3 onClick={() => history.goBack()}>Back</h3>
        </div>
    )
})

