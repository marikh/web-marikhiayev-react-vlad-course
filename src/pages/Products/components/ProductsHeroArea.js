import React from 'react';
import { HeroArea, Heading } from '../../../components';

export default () => (
    <HeroArea>
        <Heading size={3}>
            Shop
        </Heading>
    </HeroArea>
) 