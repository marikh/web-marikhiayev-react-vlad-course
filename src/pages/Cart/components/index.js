import CartHeroArea from './CartHeroArea';

export {
    CartHeroArea
}
