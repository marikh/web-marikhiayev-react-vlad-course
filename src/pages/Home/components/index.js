import HomeFooter from './HomeFooter';
import HomeHeroArea from './HomeHeroArea';

export {
    HomeFooter,
    HomeHeroArea    
}
