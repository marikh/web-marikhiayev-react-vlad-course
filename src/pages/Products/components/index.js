import ProductsHeroArea from './ProductsHeroArea';

export {
    ProductsHeroArea
}
