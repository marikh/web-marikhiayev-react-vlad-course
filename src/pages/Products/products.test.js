import { mount, shallow } from "enzyme";
import React from 'react';
import { MemoryRouter, Route, Switch, Link } from 'react-router-dom'
import { Layout, Card } from '../../components/';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux';
import ConnectedProducts,{Products} from '../../pages/Products/';
import ConnectedProductPage, {ProductPage} from '../../pages/ProductPage/';
import {createStore} from 'redux';
import { delay } from '../../common/Extensions/promises';

const MENU_INITIAL_STATE = {showProtectedLinks : false};

const products = [
    {
        id: '123123-234-2341-123123-123123',
        name: 'Greek amphora 1',
        imageUrl: '//c1.staticflickr.com/5/4215/35504896635_ec1a78af43_b.jpg',
        shortDesc: 'Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.',
        longDesc: 'Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut. Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.',
        price: '101K USD'
    },
    {
        id: '123123-12342-456456-123123-123123',
        name: 'Greek amphora 2',
        imageUrl: '//c1.staticflickr.com/5/4215/35504896635_ec1a78af43_b.jpg',
        shortDesc: 'Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.',
        longDesc: 'Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut. Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.',
        price: '102K USD'
    }];


// const PRODUCTPAGE_INITIAL_STATE = {
//   productId : "123123-234-2341-123123-123123",
//   canAddToCart : true
// };

const mockStore = configureStore()

const initialState = { allProducts: products, menu : MENU_INITIAL_STATE /*, productPage: PRODUCTPAGE_INITIAL_STATE */ } ;
    let store, wrapper;  

describe('Products outer tests', () => {

     beforeEach(()=>{
        if(wrapper != null)
            wrapper.unmount();

        store = mockStore(initialState);
        wrapper = mount( <Provider store={store}>
                            <MemoryRouter initialEntries={[ '/products' ]}>
                                <Switch>
                                    <Route render={({match}) => <ConnectedProducts props={{match: match}}/>} path="/products" />
                                    {/*<Route render={({match}) => <ConnectedProductPage props={{match: match}}/>} path="/products/:id" />*/}
                                </Switch>
                            </MemoryRouter>
                        </Provider> );
    })

    it("has Layout", () => {
        expect(wrapper.find(Layout).length).toBeGreaterThan(0)
    })
    
    it("has HeroArea Component", () => {
        expect(wrapper.find(Layout).prop('heroContent') != null).toBe(true)
    })

    it("always renders a section", () => {
        const sections = wrapper.find("section");
        expect(sections.length).toBeGreaterThan(0);
    });

    it("has 2 products Cards", () => {
        const cards = wrapper.find(Card);
        expect(cards.length).toBe(2);
    });

    it(`check good target path of product Link`, () => {
        expect(wrapper.find(Card).at(0).find(Link).at(0).prop('to')).toEqual('/products/123123-234-2341-123123-123123');
    })
})

describe('Products Shallow Render',()=>{
    let wrapper;

    beforeEach(()=>{
        wrapper = shallow(<Products products={products} match={{url: "/products"}}/>)
    })

    it('render the DUMB component', () => {
       expect(wrapper.length).toEqual(1)
    });
})


describe('Products snapshot tests', () => {

    it('test weird snapshot', () => {

        const store = mockStore(initialState);

        const tree = renderer.create(
            <Provider store={store}>
                            <MemoryRouter initialEntries={[ '/products' ]}>
                                <Route render={({match}) => <ConnectedProducts props={{match: match}}/>} path="/products" />
                            </MemoryRouter>
                        </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
 })

