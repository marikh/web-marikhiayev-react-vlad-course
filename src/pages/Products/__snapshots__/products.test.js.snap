// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products snapshot tests test weird snapshot 1`] = `
<div
  className="App"
  onClick={[Function]}
>
  <header
    className="App-header"
  >
    <div
      className="App-menubar"
    >
      <label
        className="hamburger-icon fa fa-bars"
        onClick={[Function]}
      />
      <div
        className="menu-content"
      >
        <div
          className="links-container"
        >
          <a
            className="page-link"
            href="/"
            onClick={[Function]}
          >
            <span>
              Home
            </span>
          </a>
          <a
            className="page-link"
            href="/about"
            onClick={[Function]}
          >
            <span>
              About
            </span>
          </a>
          <a
            className="page-link"
            href="/products"
            onClick={[Function]}
          >
            <span>
              Products
            </span>
          </a>
          <a
            className="page-link"
            href="/contact"
            onClick={[Function]}
          >
            <span>
              Contact
            </span>
          </a>
          <a
            className="page-link"
            href="/login"
            onClick={[Function]}
          >
            <span>
              SignIn
            </span>
          </a>
        </div>
        <div
          className="socialLinks"
        >
          <div
            className="social-links"
          >
            <a
              className="fa fa-facebook"
              href="facebook.com"
              target="_blank"
            />
            <a
              className="fa fa-github"
              href="github.com"
              target="_blank"
            />
            <a
              className="fa fa-envelope"
              href="vnovick@gmail.comm"
              target="_blank"
            />
          </div>
        </div>
      </div>
      <div
        className="overlay"
      />
    </div>
    <section
      className="App-subheader"
    >
      <div
        className="App-hero"
      >
        <h3
          className="heading heading-3"
        >
          Shop
        </h3>
      </div>
    </section>
  </header>
  <section
    className="App-body"
  >
    <div
      className="raised-2 section"
    >
      <div
        className="products-horizontal-gallery"
      >
        <div
          className="raised-2 card product-card"
          style={
            Object {
              "backgroundImage": "url(//c1.staticflickr.com/5/4215/35504896635_ec1a78af43_b.jpg)",
            }
          }
        >
          <a
            className="product-wrapping-link"
            href="/products/123123-234-2341-123123-123123"
            onClick={[Function]}
          >
            <div
              className="product-top-title"
            >
              <p
                className="product-title"
              >
                Greek amphora 1
              </p>
              <p
                className="product-price"
              >
                101K USD
              </p>
            </div>
            <p
              className="product-bottom-desc"
            >
              Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.
            </p>
          </a>
        </div>
        <div
          className="raised-2 card product-card"
          style={
            Object {
              "backgroundImage": "url(//c1.staticflickr.com/5/4215/35504896635_ec1a78af43_b.jpg)",
            }
          }
        >
          <a
            className="product-wrapping-link"
            href="/products/123123-12342-456456-123123-123123"
            onClick={[Function]}
          >
            <div
              className="product-top-title"
            >
              <p
                className="product-title"
              >
                Greek amphora 2
              </p>
              <p
                className="product-price"
              >
                102K USD
              </p>
            </div>
            <p
              className="product-bottom-desc"
            >
              Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.
            </p>
          </a>
        </div>
      </div>
    </div>
  </section>
  <footer
    className="App-footer"
  >
    <div
      className="social-links"
    >
      <a
        className="text-link"
        href="facebook.com"
        target="_blank"
      >
        Facebook
      </a>
      <a
        className="text-link"
        href="github.com"
        target="_blank"
      >
        Github
      </a>
      <a
        className="text-link"
        href="vnovick@gmail.comm"
        target="_blank"
      >
        Email
      </a>
    </div>
  </footer>
</div>
`;
